### PR TITLE
docs: add v5.1 validator upgrade guide

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Ref: #1855

This pull request adds a validator upgrade guide for `v5.1`.

Calculations: https://hackmd.io/KYpw8B01RUaeO-WsjE27tA

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: docs/validators/upgrades/v5.1-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+## Upgrade Info
+
+### Regen Ledger v5.1
+
+- [Release Notes](https://github.com/regen-network/regen-ledger/releases/tag/v5.1.0)
+
+### Regen Mainnet
+
+- Upgrade Height: `TBD`
```
Comment (ryanchristo) [inline docs/validators/upgrades/v5.1-upgrade.md:13]: TODO

File: docs/validators/upgrades/v5.1-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+### Redwood Testnet
+
+- Upgrade Height: `TBD`
```
Comment (ryanchristo) [inline docs/validators/upgrades/v5.1-upgrade.md:17]: TODO

File: docs/validators/upgrades/v5.1-upgrade.md (hunk 1 above)
Comment (ryanchristo) [reply docs/validators/upgrades/v5.1-upgrade.md]: https://github.com/regen-network/regen-ledger/pull/1869/commits/5aca1a900fc768c893b5b48b587199de14c29fb9

File: docs/validators/upgrades/v5.1-upgrade.md (hunk 2 above)
Comment (ryanchristo) [reply docs/validators/upgrades/v5.1-upgrade.md]: https://github.com/regen-network/regen-ledger/pull/1869/commits/5aca1a900fc768c893b5b48b587199de14c29fb9

File: docs/validators/upgrades/v5.1-upgrade.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+## Upgrade Info
+
+### Regen Ledger v5.1
+
+- [Release Notes](https://github.com/regen-network/regen-ledger/releases/tag/v5.1.0)
+
+### Regen Mainnet
+
+- Upgrade Height: `10068550`
```
Comment (clevinson) [inline docs/validators/upgrades/v5.1-upgrade.md:13]: Checked this is the intended height via mintscan.

**Review by clevinson — APPROVED** [review]

**Pull request merged**